examples/gno.land/p/gnoland/commondao/record.gno

```diff
@@ -0,0 +1,178 @@
+package commondao
+
+import (
+	"errors"
+	"math"
+	"std"
+
+	"gno.land/p/demo/avl"
+)
+
+// ErrVoteExists indicates that a user already voted.
+var ErrVoteExists = errors.New("user already voted")
+
+type (
+	// Vote defines a single vote.
+	Vote struct {
+		Address std.Address
+		Choice  VoteChoice
+	}
+
+	// ReadOnlyVotingRecord defines an interface for read only voting records.
+	ReadOnlyVotingRecord interface {
+		// Size returns the total number of votes that record contains.
+		Size() int
+
+		// VoteChoices returns the voting choices that has been voted.
+		VoteChoices() []VoteChoice
+
+		// Votes returns the list of all votes.
+		Votes() []Vote
```

Comment:
This might be inefficient if we have a ton of votes.

Reply:
I will change it to a `ulist` which would solve the potential issue

Reply:
I will take a look, an iterator might be the solution to the potential issue

Reply:
Addressed in 3abc2a0